src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        result.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        result.push({ ...newState });
+        break;
+
+      case 'clear':
+        newState = {};
+        result.push({ ...newState });
+    }
```

Comment:
switch should have a default case, you can throw an error if action type is invalid